pkg/k8shandler/prometheus_rule.go

```diff
@@ -21,27 +24,42 @@ const (
 )
 
 func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdatePrometheusRules() error {
-
+	ctx := context.Background()
 	dpl := elasticsearchRequest.cluster
 
-	ruleName := fmt.Sprintf("%s-%s", dpl.Name, "prometheus-rules")
+	name := fmt.Sprintf("%s-%s", dpl.Name, "prometheus-rules")
 	owner := getOwnerRef(dpl)
 
-	promRule, err := buildPrometheusRule(ruleName, dpl.Namespace, dpl.Labels)
+	rule, err := buildPrometheusRule(name, dpl.Namespace, dpl.Labels)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to build prometheus rule: %w", err)
 	}
 
-	addOwnerRefToObject(promRule, owner)
+	addOwnerRefToObject(rule, owner)
 
-	err = elasticsearchRequest.client.Create(context.TODO(), promRule)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return err
+	err = elasticsearchRequest.client.Create(ctx, rule)
```

Comment:
not sure we want to use Background context here based on the docs
`It is never canceled (sic), has no values, and has no deadline.`